From fe9ce0dabec3ea97573405de369f14acae7ced59 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Mon, 27 Nov 2023 14:44:29 +0100
Subject: [PATCH 1/9] pci: j721e: assert reset-gpio before deassertion to
 ensure reset

Optional reset gpio is already being deasserted after a 100us delay,
however no specific state has been set since devm_gpiod_get_optional.
Thus successful reset of pci card depends on a bootloader / firmware to
assert the reset line during every (re-)boot.

Explicitly assert reset-gpio to ensure reset sequence occurs independent
from boot-loader, firmware or soc reset defaults.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/pci/controller/cadence/pci-j721e-host.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/pci/controller/cadence/pci-j721e-host.c b/drivers/pci/controller/cadence/pci-j721e-host.c
index 36b1cafca7d3..7511f5dc2938 100644
--- a/drivers/pci/controller/cadence/pci-j721e-host.c
+++ b/drivers/pci/controller/cadence/pci-j721e-host.c
@@ -379,6 +379,9 @@ static int j721e_pcie_probe(struct platform_device *pdev)
 		goto err_get_sync;
 	}
 
+	if (gpiod)
+		gpiod_set_value_cansleep(gpiod, 0);
+
 	ret = cdns_pcie_init_phy(dev, cdns_pcie);
 	if (ret) {
 		dev_err(dev, "Failed to init phy\n");
-- 
2.35.3

